c7n/resources/asg.py

```diff
@@ -526,12 +526,19 @@ def get_snapshots(self, ec2, snap_ids):
             try:
                 result = ec2.describe_snapshots(SnapshotIds=snap_ids)
             except ClientError as e:
-                if e.response['Error']['Code'] == 'InvalidSnapshot.NotFound':
-                    msg = e.response['Error']['Message']
-                    e_snap_id = msg[msg.find("'") + 1:msg.rfind("'")]
-                    self.log.warning("Snapshot not found %s" % e_snap_id)
-                    snap_ids.remove(e_snap_id)
-                    continue
+                if e.response['Error']['Code'] in ['InvalidSnapshot.NotFound', 'InvalidSnapshotID.Malformed']:
```

Comment:
@salwandhruvdev this line is over the character limit and is failing travis build